### ISSUE TITLE
[Nullability] Resolve Simple variable checks

### ISSUE THREAD

**Issue by MichaelPeter** [dotnet/roslyn#45353]
Version used: 16.7.0 Preview 2.0

Not shure if here is a collector issue, but this was one case which really struck me.
```

    class Program
    {
        static void Main(string[] args)
        {
            
        }

        public void Test(MyClass myCl)
        {
            bool isSet = myCl.MyNullable != null;
            if(isSet)
            {
                Func2("a", myCl.MyNullable); // Possible null reference asssigmetn for parameter 'b'
            }
        }

        public void Func2(string a, string b)
        {

        }
    }

    public class MyClass
    {
        public string? MyNullable { get; set; }
    }

```

**Comment by jaredpar** [comment]
This behavior is "By Design". The compiler doesn't track the relationship between variables in this manner. In order to have `myC1.MyNullable` register as non-null inside the `if` block do the `null` check in the condition.

```c#
if (myC1.MyNullable != null)
```

**Comment by MichaelPeter** [comment]
This is a simplified example, but it consists only of expressions the compiler can resolve during compile time. Is it intended to keep the compiler "stupid" and not resolve known variable paths inside the same method or is it planned to make him "smarter" to show less "obvious" wrong warnings?
